### ISSUE TITLE
Unsupported product type: NST

### ISSUE THREAD

**Issue by BuserNetAdmin** [netbymatt/nexrad-level-3-data#1]
Hi Matt, I'm trying to parse a Level3 product with code 58 but I'm getting following message: 
Unsupported product type: NST

Here's the file I'm trying to parse: https://tgftp.nws.noaa.gov/SL.us008001/DF.of/DC.radar/DS.58sti/SI.kjax/sn.last

Thanks!

**Comment by netbymatt** [comment]
I haven't written a parser for that particular product yet. The readme now lists what's currently available 506000ac18ca3816e7e6bdf5b1ed831343b774ee. I looked over the definitions for NST and it looks like I can leverage parsers that already exist to write this fairly quickly. Thanks for providing sample data. Follow the nst-58 branch while I develop it.

**Comment by netbymatt** [comment]
2645fa7a41199ad431456d7779474ee7705a2ddb Adds the packet. The data is in the tabular.pages section. What I can't make sense of is the two byte string in the radialPackets[0].text. It's (quote) (space) hex codes 0x34, 0x32. Per the specification all of the data lines up on the various byte boundaries, but I don't know what those two bytes mean.

**Comment by BuserNetAdmin** [comment]
I think I maybe ended up in the same place at you, when I tried to build my own parser. I got stopped with ```Error: Unsupported packet code 0x0002```

I did confirm that MetPy can parse this file so I guess it's a good file ...

**Comment by netbymatt** [comment]
Can you provide MetPy's output for comparison. I don't do python often enough to get MetPy up and going quickly. Or I could dig into their source. I had to do that to sort out some of the original parsing code.

I'm also a little suspect of the NOAA document I'm working from. https://www.roc.noaa.gov/wsr88d/PublicDocs/ICDs/2620001Y.pdf The bottom of page 3-104 has packet code 2 but the byte-wise table that follows shows packet 1.

**Comment by netbymatt** [comment]
It is pulling out valid data. Here's the tabular data from the provided packet.
```
'[
  [
    "                            STORM POSITION/FORECAST                             ",
    "     RADAR ID 383  DATE/TIME 04:11:21/19:35:10   NUMBER OF STORM CELLS   7      ",
    "                                                                                ",
    "                   AVG SPEED 43 KTS    AVG DIRECTION 253 DEG                    ",
    "                                                                                ",
    " STORM    CURRENT POSITION              FORECAST POSITIONS               ERROR  ",
    "  ID     AZRAN     MOVEMENT    15 MIN    30 MIN    45 MIN    60 MIN    FCST/MEAN",
    "        (DEG/NM)  (DEG/KTS)   (DEG/NM)  (DEG/NM)  (DEG/NM)  (DEG/NM)     (NM)   ",
    "                                                                                ",
    "  L9     273/156     NEW       NO DATA   NO DATA   NO DATA   NO DATA    0.0/ 0.0",
    "  J7     140/223   236/ 30     138/224   136/225   NO DATA   NO DATA    1.0/ 1.1",
    "  P7     263/160     NEW       NO DATA   NO DATA   NO DATA   NO DATA    0.0/ 0.0",
    "  H0     188/ 95     NEW       NO DATA   NO DATA   NO DATA   NO DATA    0.0/ 0.0",
    "  B1     181/ 86     NEW       NO DATA   NO DATA   NO DATA   NO DATA    0.0/ 0.0",
    "  I2      96/ 98     NEW       NO DATA   NO DATA   NO DATA   NO DATA    0.0/ 0.0",
    "  M1     182/ 69   262/ 57     NO DATA   NO DATA   NO DATA   NO DATA    3.6/ 2.5"
  ],
  [
    "              STORM CELL TRACKING/FORECAST ADAPTATION DATA                      ",
    "                                                                                ",
    "    225   (DEG) DEFAULT (DIRECTION)      2.5   (M/S) THRESH (MINIMUM SPEED)     ",
    "   25.0   (KTS) DEFAULT (SPEED)           20    (KM) ALLOWABLE ERROR            ",
    "     20   (MIN) TIME (MAXIMUM)            15   (MIN) FORECAST INTERVAL          ",
    "     10         NUMBER OF PAST VOLUMES     4         NUMBER OF INTERVALS        ",
    "   30.0   (M/S) CORRELATION SPEED         15   (MIN) ERROR INTERVAL             ",
    "                                                                                ",
    "                                                                                ",
    "              SCIT REFLECTIVITY MEDIAN FILTER                                   ",
    "                                                                                ",
    "    7.0   (KM)  FILTER KERNEL SIZE       0.5         THRESH (FILTER FRACTION)   ",
    "    Yes         REFLECTIVITY FILTERED                                           "
  ]
]'
```

**Comment by netbymatt** [comment]
I tracked down the documentation for the two byte string. It is valid and it relates to a table in another document that defines what symbol should be used to show the storms on a graphic display (current position, forecast position, previous position, etc). Thus the parsing implemented with f8c41cadd733c456302806865c87343061949715 is correct and the interpretation of those special characters would be up to the display mechanism if needed. It sounds like the same information can also be determined from the tabular data.
